diff --git a/config.def.h b/config.def.h
index 9efa774..6ba063c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -2,6 +2,7 @@
 
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static const unsigned int gappx     = 1;        /* gap pixel between windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
diff --git a/dwm.c b/dwm.c
index 0fbf43a..51ae94b 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1696,32 +1696,35 @@ tagmon(const Arg *arg)
 void
 tile(Monitor *m)
 {
-	unsigned int i, n, h, mw, my, ty, bw;
+	unsigned int i, n, h, mw, my, ty, bw, g;
 	Client *c;
 
 	for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
 	if (n == 0)
 		return;
 
-	if (n == 1)
+	if (n == 1) {
 		bw = 0;
-	else
+		g = 0;
+	} else {
 		bw = borderpx;
+		g = gappx;
+	}
 	if (n > m->nmaster)
-		mw = m->nmaster ? m->ww * m->mfact : 0;
+ 		mw = m->nmaster ? (m->ww - g) * m->mfact : 0;
 	else
-		mw = m->ww;
-	for (i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
+ 		mw = m->ww - g;
+ 	for (i = 0, my = ty = g, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
 		if (i < m->nmaster) {
-			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
- 			resize(c, m->wx, m->wy + my, mw - 2*bw, h - 2*bw, bw, 0);
-			if (my + HEIGHT(c) < m->wh)
-				my += HEIGHT(c);
+ 			h = (m->wh - my) / (MIN(n, m->nmaster) - i) - g;
+ 			resize(c, m->wx + g, m->wy + my, mw - 2*bw - g, h - 2*bw, bw, 0);
+			if (my + HEIGHT(c) + g < m->wh)
+				my += HEIGHT(c) + g;
 		} else {
-			h = (m->wh - ty) / (n - i);
- 			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - 2*bw, h - 2*bw, bw, 0);
-			if (ty + HEIGHT(c) < m->wh)
-				ty += HEIGHT(c);
+ 			h = (m->wh - ty) / (n - i) - g;
+ 			resize(c, m->wx + mw + g, m->wy + ty, m->ww - mw - 2*bw - 2*g, h - 2*bw, bw, 0);
+			if (ty + HEIGHT(c) + g < m->wh)
+				ty += HEIGHT(c) + g;
 		}
 }
 
